src/v/rpc/transport.h

```diff
@@ -103,18 +104,31 @@ class transport final : public base_transport {
 
 private:
     friend client_context_impl;
-
     ss::future<> do_reads();
     ss::future<> dispatch(header);
     void fail_outstanding_futures() noexcept final;
     void setup_metrics(const std::optional<ss::sstring>&);
 
+    ss::future<result<std::unique_ptr<streaming_context>>>
+      do_send(uint32_t, netbuf, rpc::client_opts);
+    void dispatch_send();
+
+    result<internal::response_handler*>
+    assign_correlation_id(const rpc::client_opts&);
+
     ss::semaphore _memory;
     absl::flat_hash_map<uint32_t, std::unique_ptr<internal::response_handler>>
       _correlations;
     uint32_t _correlation_idx{0};
     ss::metrics::metric_groups _metrics;
-
+    /**
+     * ordered map containing in-flight requests. The map preserves order of
+     * calling send_typed function. It is fine to use btree_map in here as it
+     * ususally contains only few elements.
```

Comment:
can you 'measure' what is 'few' i think should be in the 10's of elements

Reply:
We have a metric saying requests pending in the client it is usually less than 20.